### PR TITLE
Navbar 

### LINKED ISSUE #2: Navigation menu
- The web application must have a menu at the top.
- The menu should contain a button with the "Start" option.
- When selecting the "Start" option in the menu, the application should redirect the user to a new screen.

On this screen, all recorded events should be displayed.

For each event, the following details must be displayed:

Title: The title of the event.
Description: The description of the event.
Date: The date of the event.
Location: The location of the event.
Cost: The cost of the event.

### PR DESCRIPTION
**Pull request by JoseLopezSolis**
We have created the "event" controller which is the controller to manage our events, also we created the navbar that shows a button to redirect the user to another view called "show", this view was created as well, and on this view, the user will be able to see the events that are stored in the database.

For more details see #2 
Screenshot 2023-06-28 183023
Screenshot 2023-06-28 182952

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -13,4 +13,20 @@
  *= require_tree .
  *= require_self
  */
- @import "bootstrap";
+@import "bootstrap";
+/* Base */
```
Comment (cfcortesmx) [inline app/assets/stylesheets/application.scss:18]: Consider adding a comment to explain what this section does.

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -13,4 +13,20 @@
...
+body {
+  padding: 0;
+  margin: 0;
+  h1 {
+    color: hsl(0, 0%, 28%);
+    font-size: 25px;
+    font-weight: bold;
+  }
+  .hover-span:hover {
```
Comment (cfcortesmx) [inline app/assets/stylesheets/application.scss]: Consider adding a comment to explain what this section does.

File: app/assets/stylesheets/application.scss [hunk 3]
```diff
@@ -13,4 +13,20 @@
...
+    border-radius: 50%;
+  }
+  .hover-span:hover {
```
Comment (cfcortesmx) [inline app/assets/stylesheets/application.scss:30]: Consider adding a comment to explain what this section does.

File: app/controllers/events_controller.rb [hunk 4]
```diff
@@ -1,3 +1,10 @@
 class EventsController < ApplicationController
-  def index; end
-end
+  def index
```
Comment (cfcortesmx) [inline app/controllers/events_controller.rb:3]: Consider adding a comment to describe the purpose of the 'index' action.

File: app/controllers/events_controller.rb [hunk 5]
```diff
@@ -1,3 +1,10 @@
...
+    @events = Event.all
+  end
+
+  def show
```
Comment (cfcortesmx) [inline app/controllers/events_controller.rb:7]: Consider adding a comment to describe the purpose of the 'show' action.

File: app/models/event.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+class Event < ApplicationRecord
```
Comment (cfcortesmx) [inline app/models/event.rb:1]: Consider adding some validations to the Event model to ensure data integrity.

File: app/views/events/index.html.erb [hunk 7]
```diff
@@ -1,5 +1,4 @@
-<div class="container">
-  <div class="row text-center">
+<div class="container d-flex justify-content-center">
```
Comment (cfcortesmx) [inline app/views/events/index.html.erb:1]: Consider removing the unnecessary div tag.

File: app/views/events/index.html.erb [hunk 8]
```diff
@@ -1,5 +1,4 @@
...
     <h1>Events Management System</h1>
```
Comment (cfcortesmx) [inline app/views/events/index.html.erb:2]: Consider removing the unnecessary div tag.

File: app/views/events/index.html.erb [hunk 9]
```diff
@@ -1,5 +1,4 @@
...
-  </div>
 </div>
```
Comment (cfcortesmx) [inline app/views/events/index.html.erb:3]: Consider removing the unnecessary div tag.

File: app/views/events/index.html.erb [hunk 10]
```diff
@@ -1,5 +1,4 @@
...
+
```
Comment (cfcortesmx) [inline app/views/events/index.html.erb]: Consider removing the unnecessary div tag.

File: app/views/events/show.html.erb [hunk 11]
```diff
@@ -0,0 +1,47 @@
+<div class="container">
```
Comment (cfcortesmx) [inline app/views/events/show.html.erb:1]: Consider adding a comment to describe the purpose of this file.

File: app/views/events/show.html.erb [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+  <h1>Events</h1>
```
Comment (cfcortesmx) [inline app/views/events/show.html.erb:2]: Consider adding a comment to describe the purpose of this section.

File: app/views/events/show.html.erb [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+
+<table class="table table-striped table-hover ">
```
Comment (cfcortesmx) [inline app/views/events/show.html.erb:4]: Consider adding a comment to describe the purpose of this table.

File: app/views/events/show.html.erb [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+  <thead class="table-dark"> 
+    <tr>
```
Comment (cfcortesmx) [inline app/views/events/show.html.erb:6]: Consider adding a comment to describe the purpose of this table header.

File: app/views/events/show.html.erb [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+      <th scope="col">ID</th>
+      <th scope="col">Title</th>
+      <th scope="col">Description</th>
+      <th scope="col">Date</th>
+      <th scope="col">Location</th>
+      <th scope="col">Cost</th>
+      <th colspan="2" scope="col">Actions</th>
```
Comment (cfcortesmx) [inline app/views/events/show.html.erb:13]: Consider adding a comment to describe the purpose of this table header.

File: app/views/events/show.html.erb [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+    </tr>
+  </thead>
+  <tbody>
+    <% @events.each do |event| %>
```
Comment (cfcortesmx) [inline app/views/events/show.html.erb]: Consider adding a comment to describe the purpose of this loop.

File: app/views/events/show.html.erb [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+          <%= event.date %>
+        </td>
+        <td>
+          <%= event.location %>
+        </td>
+        <td>
+          $<%= event.cost %>
+        </td>
+        <td>
+          <span class="material-symbols-outlined hover-span"> edit </span>
```
Comment (cfcortesmx) [inline app/views/events/show.html.erb:39]: Consider using a button element instead of a span for better accessibility.

File: app/views/events/show.html.erb [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+        </td>
+        <td>
+          <span class="material-symbols-outlined hover-span"> delete </span>
```
Comment (cfcortesmx) [inline app/views/events/show.html.erb:42]: Consider using a button element instead of a span for better accessibility.

File: app/views/layouts/application.html.erb [hunk 19]
```diff
@@ -7,12 +7,13 @@
     <meta name="viewport" content="width=device-width,initial-scale=1">
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
+    <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@48,400,0,0" />
```
Comment (cfcortesmx) [inline app/views/layouts/application.html.erb]: Consider moving the Google Fonts link to the head section of the HTML document for better performance.

File: app/views/navbar/_navbar.html.erb [hunk 20]
```diff
@@ -0,0 +1,11 @@
+<nav class="navbar navbar-expand-lg navbar-light bg-light">
```
Comment (cfcortesmx) [inline app/views/navbar/_navbar.html.erb:1]: Consider adding a comment to describe the purpose of this file.

File: app/views/navbar/_navbar.html.erb [hunk 21]
```diff
@@ -0,0 +1,11 @@
...
+    <div class="container">
+      <a class="navbar-brand" href="/">Brighcoders Events</a>
```
Comment (cfcortesmx) [inline app/views/navbar/_navbar.html.erb:3]: Consider adding a comment to describe the purpose of this element.

File: app/views/navbar/_navbar.html.erb [hunk 22]
```diff
@@ -0,0 +1,11 @@
...
+      <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
+        <span class="navbar-toggler-icon"></span>
+      </button>
+      <div class="collapse navbar-collapse justify-content-end" id="navbarNav">
+        <a class="btn btn-primary ms-lg-3" href="/show">Iniciar</a>
```
Comment (cfcortesmx) [inline app/views/navbar/_navbar.html.erb]: Consider adding a comment to describe the purpose of this link.

File: config/routes.rb [hunk 23]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  get '/events', to: 'events#index'
+  get '/', to: 'events#index'
```
Comment (cfcortesmx) [inline config/routes.rb]: Consider using a more descriptive route name instead of '/'.

File: config/routes.rb [hunk 24]
```diff
@@ -1,3 +1,4 @@
...
+  get '/show', to: 'events#show'
```
Comment (cfcortesmx) [inline config/routes.rb]: Consider using a more descriptive route name instead of '/show'.

File: test/models/event_test.rb [hunk 25]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (cfcortesmx) [inline test/models/event_test.rb:1]: Consider adding a comment to describe the purpose of this file.

File: app/controllers/events_controller.rb [hunk 26]
```diff
@@ -1,3 +1,10 @@
...
+    @events = Event.all
+  end
+
+end
```
Comment (Alonso2105) [inline app/controllers/events_controller.rb]: add end line please, if you are using `VScode` as editor you can leave this setting automatic on save
Screenshot 2023-06-28 at 22 52 49

[EOL Documentation](https://instrumentationtools.com/use-end-line-eol-resistor-fire-gas-system/#:~:text=What%20are%20End%20of%20Line%20Resistors%20and%20do%20I%20need%20them%20on%20my%20alarm%20system%3F)

File: app/views/navbar/_navbar.html.erb (hunk 22 above)
Comment (Alonso2105) [inline app/views/navbar/_navbar.html.erb]: please be consistent with the system, if it is in English or Spanish 👁️ and don't forget that your requirement is specific **[The menu should contain a button with the "Start" option.](https://github.com/BrightCoders-Institute/reto-administrador-de-eventos-jdg/issues/2#:~:text=The%20menu%20should%20contain%20a%20button%20with%20the%20%22Start%22%20option.)**

File: app/controllers/events_controller.rb (hunk 5 above)
Comment (Alonso2105) [inline app/controllers/events_controller.rb:7]: Checking the official rails documentation, the show method should be used to display the detail of an object, not the list of it, please consider following good practices andmaking this change, attached documentation
[Rails documentation](https://guides.rubyonrails.org/routing.html#crud-verbs-and-actions)

File: app/views/navbar/_navbar.html.erb [hunk 27]
```diff
@@ -0,0 +1,11 @@
...
+      </div>
+    </div>
+</nav>
```
Comment (Alonso2105) [inline app/views/navbar/_navbar.html.erb]: same here, add end line please

File: config/routes.rb (hunk 24 above)
Comment (Alonso2105) [inline config/routes.rb]: show method should return the detail of an object, check the official documentation for this modification please
[Rails Documentation](https://guides.rubyonrails.org/routing.html#connecting-urls-to-code)

File: test/models/event_test.rb [hunk 28]
```diff
@@ -0,0 +1,7 @@
...
+
+class EventTest < ActiveSupport::TestCase
+  # test "the truth" do
```
Comment (Alonso2105) [inline test/models/event_test.rb:4]: please add model tests
https://guides.rubyonrails.org/testing.html#rails-meets-minitest

File: app/assets/stylesheets/application.scss [hunk 29]
```diff
@@ -13,4 +13,20 @@
  *= require_tree .
  *= require_self
  */
- @import "bootstrap";
+@import "bootstrap";
+/* Base */
+body {
+  padding: 0;
+  margin: 0;
```
Comment (juperameza) [inline app/assets/stylesheets/application.scss]: Order this properties alphabetically.

Comment (juperameza) [reply app/assets/stylesheets/application.scss]: And try to add some structure to the css for  better usage on the future

**Review by juperameza — CHANGES_REQUESTED** [review]
Resolve Alonso conversations. And mine

File: test/models/event_test.rb (hunk 28 above)
Comment (JoseLopezSolis) [reply test/models/event_test.rb:4]: We're going to add the model test in the next issue, thanks for the observation.

File: app/assets/stylesheets/application.scss [hunk 30]
```diff
@@ -13,4 +13,21 @@
  *= require_tree .
  *= require_self
  */
- @import "bootstrap";
+//Importing bootstrap
```
Comment (cfcortesmx) [inline app/assets/stylesheets/application.scss:16]: Consider moving the import statement for bootstrap to the top of the file for better organization and readability.

File: app/assets/stylesheets/application.scss [hunk 31]
```diff
@@ -13,4 +13,21 @@
...
+@import "bootstrap";
+/* Base */
+:root {
```
Comment (cfcortesmx) [inline app/assets/stylesheets/application.scss:19]: Consider using CSS variables for defining colors and font sizes. This will make it easier to update the styles in the future and maintain consistency throughout the application.

Comment (cfcortesmx) [inline app/assets/stylesheets/application.scss:19]: There seems to be a typo in the variable name. It should be '--header-color' instead of '--head-color'.

File: app/assets/stylesheets/application.scss [hunk 32]
```diff
@@ -13,4 +13,21 @@
...
+  --header-color: hsl(0, 0%, 28%);
+  --font-size-header: 25px;
+}
+body {
+  h1 {
+    color: var(--head-color);
+    font-size: var(--font-size-header);
```
Comment (cfcortesmx) [inline app/assets/stylesheets/application.scss:26]: Remove the commented out code. It's unnecessary and can clutter the codebase.

File: app/controllers/events_controller.rb [hunk 33]
```diff
@@ -1,3 +1,7 @@
 class EventsController < ApplicationController
-  def index; end
+  #This method will show the events in /events view
```
Comment (cfcortesmx) [inline app/controllers/events_controller.rb:2]: Consider renaming the method `events` to `index` to follow Rails conventions for controller actions.

File: app/controllers/events_controller.rb [hunk 34]
```diff
@@ -1,3 +1,7 @@
...
+  def events
```
Comment (cfcortesmx) [inline app/controllers/events_controller.rb]: Remove the comment `#This method will show the events in /events view` as it is redundant and does not provide any additional information.

File: app/controllers/events_controller.rb [hunk 35]
```diff
@@ -1,3 +1,7 @@
...
+    @events = Event.all
+  end
+
```
Comment (cfcortesmx) [inline app/controllers/events_controller.rb:6]: Remove the unused `show` action as it is not being used in this controller.

File: app/models/event.rb [hunk 36]
```diff
@@ -0,0 +1,7 @@
...
+  validates :title , presence:true
```
Comment (cfcortesmx) [inline app/models/event.rb:2]: Consider adding a blank line after the class declaration for better readability.

File: app/models/event.rb [hunk 37]
```diff
@@ -0,0 +1,7 @@
...
+  validates :description , presence:true
```
Comment (cfcortesmx) [inline app/models/event.rb:3]: Consider adding a blank line before the first validation for better readability.

File: app/models/event.rb [hunk 38]
```diff
@@ -0,0 +1,7 @@
...
+  validates :date , presence:true
```
Comment (cfcortesmx) [inline app/models/event.rb:4]: Consider adding a blank line between each validation for better readability.

File: app/views/events/events.html.erb [hunk 39]
```diff
@@ -0,0 +1,48 @@
...
+      <th scope="col">Title</th>
+      <th scope="col">Description</th>
+      <th scope="col">Date</th>
+      <th scope="col">Location</th>
+      <th scope="col">Cost</th>
+      <th colspan="2" scope="col">Actions</th>
+    </tr>
+  </thead>
+  <tbody>
+  <!-- In this loop we're showing the events-->
```
Comment (cfcortesmx) [inline app/views/events/events.html.erb]: Consider adding a comment to explain the purpose of this loop.

File: app/views/events/index.html.erb [hunk 40]
```diff
@@ -1,5 +1,3 @@
-<div class="container">
-  <div class="row text-center">
+<div class="container d-flex justify-content-center">
```
Comment (cfcortesmx) [inline app/views/events/index.html.erb:1]: Consider removing the extra space at the end of the line.

File: app/views/layouts/application.html.erb [hunk 41]
```diff
@@ -3,6 +3,7 @@
   <head>
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>  
     <%= javascript_importmap_tags %>
+    <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@48,400,0,0" />
```
Comment (cfcortesmx) [inline app/views/layouts/application.html.erb:6]: Consider moving the Google Fonts stylesheet link to the head section of the HTML document for better performance. This will ensure that the font styles are loaded before rendering the page.

File: app/views/navbar/_navbar.html.erb [hunk 42]
```diff
@@ -0,0 +1,11 @@
+<nav class="navbar navbar-expand-lg navbar-light bg-light">
+    <div class="container">
+      <a class="navbar-brand" href="/">Brighcoders Events</a>
+      <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
+        <span class="navbar-toggler-icon"></span>
+      </button>
+      <div class="collapse navbar-collapse justify-content-end" id="navbarNav">
+        <a class="btn btn-primary ms-lg-3" href="/events">Start</a>
```
Comment (cfcortesmx) [inline app/views/navbar/_navbar.html.erb]: The button text should be more descriptive and meaningful. Instead of 'Start', consider using 'View Events' or something similar.

File: config/routes.rb [hunk 43]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  get '/events', to: 'events#index'
+  root to: 'events#index'
```
Comment (cfcortesmx) [inline config/routes.rb]: Consider using the `root` method instead of `get '/'` to define the root route. This provides a more explicit and readable way to define the root route.

File: config/routes.rb [hunk 44]
```diff
@@ -1,3 +1,4 @@
...
+  get '/events', to: 'events#events'
```
Comment (cfcortesmx) [inline config/routes.rb]: The route '/show' seems unnecessary. If the purpose is to show events, it would be better to use a RESTful route like '/events/:id' to show a specific event. Consider removing this route and using a show action within the 'events' controller instead.

File: config/routes.rb [hunk 45]
```diff
@@ -1,3 +1,4 @@
...
 end
```
Comment (cfcortesmx) [inline config/routes.rb:5]: The route '/events' should be named more descriptively. Consider using a more meaningful name like '/events/:id' to show a specific event or '/events/new' to create a new event.

File: app/controllers/events_controller.rb [hunk 46]
```diff
@@ -1,3 +1,7 @@
 class EventsController < ApplicationController
-  def index; end
+  #This method will show the events in /events view
+  def events
+    @events = Event.all
+  end
```
Comment (juperameza) [inline app/controllers/events_controller.rb]: Why did you delete the restful methods?

**Review by juperameza — CHANGES_REQUESTED** [review]

File: app/controllers/events_controller.rb (hunk 46 above)
Comment (JoseLopezSolis) [reply app/controllers/events_controller.rb]: The index method didn't do anything, instead the events method does the request of the data, and show it in events view.

File: app/views/navbar/_navbar.html.erb [hunk 47]
```diff
@@ -5,7 +5,7 @@
         <span class="navbar-toggler-icon"></span>
       </button>
       <div class="collapse navbar-collapse justify-content-end" id="navbarNav">
-        <a class="btn btn-primary ms-lg-3" href="/events">Start</a>
+        <a class="btn btn-primary" href="/events/show">START</a>
```
Comment (juperameza) [inline app/views/navbar/_navbar.html.erb:8]: Use link to in rails

**Review by juperameza — APPROVED** [review]

File: app/controllers/events_controller.rb (hunk 46 above)
Comment (juperameza) [reply app/controllers/events_controller.rb]: its better to do restful methods instead of that events its better to do index

**Review by JaimeGonz — APPROVED** [review]
Well done!
